common/sgx/eeid.c

```diff
@@ -356,19 +356,21 @@ oe_result_t oe_remeasure_memory_pages(
 
     if (with_eeid_pages)
     {
+        char* eeid_bytes = (char*)eeid;
         size_t num_bytes = oe_eeid_byte_size(eeid);
         size_t num_pages =
             num_bytes / OE_PAGE_SIZE + ((num_bytes % OE_PAGE_SIZE) ? 1 : 0);
 
-        oe_page_t* pages = malloc(sizeof(oe_page_t) * num_pages);
-        memset(pages, 0, sizeof(oe_page_t) * num_pages);
-        memcpy(pages->data, eeid, num_bytes);
-
+        oe_page_t* page = malloc(sizeof(oe_page_t));
         for (size_t i = 0; i < num_pages; i++)
-            OE_CHECK(
-                _add_page(&hctx, base, pages + i * OE_PAGE_SIZE, &vaddr, true));
-
-        free(pages);
+        {
+            memset(page->data, 0, sizeof(oe_page_t));
+            size_t n = (i != num_pages - 1) ? OE_PAGE_SIZE
+                                            : (num_bytes % OE_PAGE_SIZE);
+            memcpy(page->data, eeid_bytes + OE_PAGE_SIZE * i, n);
+            OE_CHECK(_add_page(&hctx, base, page->data, &vaddr, true));
+        }
+        free(page);
```

Comment:
For symmetry with the use of `oe_memalign` (the special free is so that common code has the appropriate semantics on Windows as well, which requires `_aligned_free` for `_aligned_malloc`)
```suggestion
        oe_memalign_free(page);
```